### PR TITLE
Fixed grammar, updated to only ask for required permissions

### PR DESCRIPTION
**Pull request by flash76**
## Description

I updated the code to only ask for permissions that are required so far. I haven't tested it thoroughly, so the app may crash, but that can easily be fixed, and fixed some grammar in the Strings.

Fixes #(issue)

## Type of change
Just put an x in the [] which are valid.
- [ x ] Bug fix (non-breaking change which fixes an issue)
- [    ] New feature (non-breaking change which adds functionality)
- [    ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [ x ] `./gradlew assembleDebug assembleRelease`
- [ ] `./gradlew checkstyle`

# Checklist:
- [ x ] My code follows the style guidelines of this project
- [ x ] I have performed a self-review of my own code
- [ x ] I have commented my code, particularly in hard-to-understand areas
- [ x ] I have made corresponding changes to the documentation
- [ x ] My changes generate no new warnings

### CODE REVIEW

File: Android/app/src/main/java/io/github/project_travel_mate/travel/ShoppingCurrentCityActivity.java [hunk 1]
```diff
@@ -87,6 +91,21 @@ protected void onCreate(Bundle savedInstanceState) {
 
         setTitle(getResources().getString(R.string.text_shopping));
 
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/travel/ShoppingCurrentCityActivity.java]: too much code repitition.  Can there be a simpler way for this?

Comment (flash76) [reply Android/app/src/main/java/io/github/project_travel_mate/travel/ShoppingCurrentCityActivity.java]: I _could_ create a public method that takes the permission of a parameter and requests it if not already, and call that in the required places.

Comment (Swati4star) [reply Android/app/src/main/java/io/github/project_travel_mate/travel/ShoppingCurrentCityActivity.java]: Let's have a new interface to request for specific permissions and our class can implement whichever needed. "Interface segregation principle" can be applied here. Let me know if you need any help here.

Comment (flash76) [reply Android/app/src/main/java/io/github/project_travel_mate/travel/ShoppingCurrentCityActivity.java]: What do you mean by interface? Should I create methods that a class will only implement if required?

Comment (prabhakar267) [reply Android/app/src/main/java/io/github/project_travel_mate/travel/ShoppingCurrentCityActivity.java]: @flash76 Interface is a blueprint for a class, you can declare methods in interfaces and let the class implementation to it implement the actual method.
You can read about interfaces here: https://www.geeksforgeeks.org/interfaces-in-java/

**Pull request closed without merging**